spec.html

```diff
@@ -13625,6 +13650,31 @@ <h1>Static Semantics: AssignmentTargetType</h1>
           1. Return ~invalid~.
         </emu-alg>
       </emu-clause>
+
+      <emu-clause id="sec-callexpression-static-semantics-presentinstacktraces">
+        <h1>Static Semantics: PresentInStackTraces</h1>
+        <emu-see-also-para op="PresentInStackTraces"></emu-see-also-para>
+        <emu-grammar>CallExpression : CoverCallExpressionAndAsyncArrowHead</emu-grammar>
+        <emu-alg>
+          1. If |CallExpression| occurs within a |ScriptBody|, |ModuleBody|, or |FunctionBody| that has a Directive Prologue that contains a Sensitive Directive, return *false*.
+          1. If the source text matched by |CallExpression| is eval code resulting from a direct eval, then
+            1. Return PresentInStackTraces of the |CallExpression| whose evaluation is the direct eval.
```

Comment:
I think I get what you're trying to do here, but magically rematerializing the instance of the ancestor |CallExpression| that resulted in the particular direct eval that led to the instance of this |CallExpression| is harder to think about than I'd like. I wonder if there's a way to write this in a more obvious top-down manner, but the interleaving of static and run-time semantics definitely makes that very challenging.

One option is to have an argument-taking Static Semantics where direct eval can pass in the inherited "is sensitive" or "hide source" state. Not sure if that's a net editorial improvement...

Reply:
> One option is to have an argument-taking Static Semantics where direct eval can pass in the inherited "is sensitive" or "hide source" state. Not sure if that's a net editorial improvement...

The problem is that we need the relevant bit at this point, during runtime semantics. Having it at the point that you are doing the `eval` call doesn't really help unless you propagate that information on some runtime construct like lexical environments.

Reply:
Hm, I see. If we can't come up with more obvious verbiage here, I think that sentence warrants some note that all this is doing is trying to reflect the "lexical" nesting of the directives across (space and?) time, due to direct evals.

Reply:
We could add a NOTE reading something like

> NOTE: Sensitive Directives apply to nested functions, including to code within a direct call to `eval`.

I'm not in love with that wording but don't have a better one off the top of my head.